contracts/src/Gateway.sol

```diff
@@ -76,19 +77,23 @@ contract Gateway is IGateway, IInitializable {
         address agentExecutor,
         uint256 dispatchGas,
         ParaID bridgeHubParaID,
-        bytes32 bridgeHubHubAgentID,
+        bytes32 bridgeHubAgentID,
         ParaID assetHubParaID,
-        bytes32 assetHubHubAgentID,
+        bytes32 assetHubAgentID,
         bytes2 createTokenCallID
     ) {
+        if (bridgeHubParaID == assetHubParaID || bridgeHubAgentID == assetHubAgentID) {
```

Comment:
Lets also check that none of these variables are zero:
* AgentIDs
* ParaIDs
* dispatchGas